### PR TITLE
Rewrite of the pack publishing and tutorial sections

### PR DESCRIPTION
**Pull request by KeilChris**

### CODE REVIEW

File: doxygen/images/pack_trinity.png
Comment (jkrech) [inline doxygen/images/pack_trinity.png:1]: I like trinity but I think we must not forget about projects:
- examples
- templates
- reference applications

File: doxygen/src/pdsc_format.txt [hunk 1]
```diff
@@ -447,7 +447,7 @@ this example applies to \ref element_release.
 /**
 \page element_package_description /package/description element
 
-Contains a brief description of the \ref cp_SWComponents "Software Pack". Optionally with link to an file with overview documentation in Markdown format.
+Contains a brief description of the software pack. Optionally with link to an file with overview documentation in Markdown format.
```
Comment (jkrech) [inline doxygen/src/pdsc_format.txt:450]: to a file

File: doxygen/src/pdsc_format.txt [hunk 2]
```diff
@@ -295,13 +295,13 @@ assists searching for packages.
...
     <td>\ref element_keywords "keywords"</td>
-    <td>Defines keywords that might be used to find a \ref cp_SWComponents "Software Pack". Keywords may be also exported to web page to provide information for search engines, such as Google.</td>
+    <td>Defines keywords that might be used to find a software pack. Keywords may be also exported to web page to provide information for search engines, such as Google.</td>
     <td>group</td>
     <td>0..1 </td>
   </tr>
   <tr>
     <td>\ref element_generators "generators"</td>
-    <td>Specifies generator tools that have been used to create the \ref cp_SWComponents "Software Pack".</td>
+    <td>Specifies generator tools that have been used to create the software pack.</td>
```
Comment (jkrech) [inline doxygen/src/pdsc_format.txt:304]: This does not sound right. This is the classic generator which generate a gpdsc (not pdsc) that is included into the project. What am I missing? Not sure when this got introduced.

File: doxygen/src/pdsc_format.txt
Comment (jkrech) [inline doxygen/src/pdsc_format.txt:1]: looks like we do not create a reference to "software pack" anymore. Can you check who introduced it?

File: doxygen/src/pack_publish.txt [hunk 3]
```diff
@@ -1,75 +1,225 @@
...
+
+\note For automatic updates on a local machine, specify the \c \elem{url} using the
+<a href="https://en.wikipedia.org/wiki/File_URI_scheme" target="_blank">file URI scheme</a>.
 
-\image html LocalInstallation.png "Local installation and update from URL"
+\section cp_WebDownload Web download
 
-\note For automatic updates on a local machine, specify the \c \<url\> using the
-<a href="https://en.wikipedia.org/wiki/File_URI_scheme" target="_blank">file URI scheme</a>.
+If you want to make your packs widely available, you need to host them on the web. For publishing packs, these
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:44]: widely <-> publicly

File: doxygen/src/pack_publish.txt [hunk 4]
```diff
@@ -1,75 +1,225 @@
...
+options are available:
+- \ref cp_HowWebDownloadWorks
+- \ref cp_GitHubHosting
+
+The following diagram shows how you can publish your packs when using a \ref cp_KeilComPack "pack indexing service":
+
+\image html IndexingService.png "Pack hosting and indexing service"
+
+When you run the command \c cpackget \c update-index, the tool queries the \ref pidxFile "package index file (index.pidx)"
+from the pack index server. This file is generated using the \ref cp_vidx2pidx "vidx2pidx tool" from the
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:54]: Unfortunately we don't run vidx2pidx on the keil.com server.
Also I am not aware that vidx2pidx would run any schema check on pdsc files.
The vidx2pidx is really an offline utility to create dedicated index.pidx files based on a custom `*.vidx` file.

File: doxygen/src/pack_publish.txt [hunk 5]
```diff
@@ -1,75 +1,225 @@
...
+\ref vidxFile "vendor index file (index.vidx)" which points to all known \elem{vendor}.pidx and stand-alone PDSC files.
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:55]: all known -> a list of included/watched/...

File: doxygen/src/pack_publish.txt [hunk 6]
```diff
@@ -1,75 +1,225 @@
...
-Any web server can be used to host a software pack (specified by the \c \<url\> element in the PDSC file). At this location,
+Any web server can be used to host a software pack (specified by the \c \elem{url} element in the PDSC file). At this location,
 the following files should be present:
 -# \elem{vendor}.\elem{name}.pdsc [required]: pack description file.
 -# \elem{vendor}.\elem{name}.\elem{version}.pack [required]: pack file where \elem{version} refers to the latest version
    specified in the PDSC file.
 -# \elem{vendor}.pidx [optional]: list with all packs hosted and maintained by the vendor (refer to \ref pidxFile)
 
-All previous versions listed in the \elem{releases} section of the PDSC file should be present at the \c \<url\> as well. This
+All previous versions listed in the \elem{releases} section of the PDSC file should be present at the \c \elem{url} as well. This
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:71]: Should we add the rule here that pack versions must not be removed?!

File: doxygen/src/pack_publish.txt [hunk 7]
```diff
@@ -1,75 +1,225 @@
...
+  <url>https://github.com/orgname/indexrepo/raw/main/</url>
+  <timestamp>2024-08-22 10:08:36</timestamp>
+  <pindex>
+    <pdsc url="https://github.com/orgname/name1/raw/main/" vendor=”orgname" name=”name1" version="1.0.0"/>
+    <pdsc url="https://github.com/orgname/name2/raw/main/" vendor=”orgname" name=”name2" version="1.0.0"/>
+  </pindex>
+</index>
+\endcode
+
+\note When updating one of the packs, you only need to change the version number. All other entries remain the same.
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:119]: Update the timestamp as well.

File: doxygen/src/pack_publish.txt [hunk 8]
```diff
@@ -1,75 +1,225 @@
...
+
+\code{.xml}
+<?xml version="1.0" encoding="utf-8"?>
+
+<package schemaVersion="1.7.7" xmlns:xs="http://www.w3.org/2001/XMLSchema-instance" xs:noNamespaceSchemaLocation="PACK.xsd">
+  <vendor>orgname</vendor>
+  <name>name1</name>
+  <license>LICENSE</license>
+  <description>Whatever Pack</description>
+  <url>https://github.com/orgname/name1/releases/download/v1.0.0/</url>
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:133]: The <url> element in the pdsc file must remain constant and must not have a version!
We had agreed that this should point to the raw file in the main branch except that the file will be composed from `<vendor>.<name>.pdsc.`
<url>https://github.com/orgname/name1/raw/main/</url>

File: doxygen/src/pack_publish.txt [hunk 9]
```diff
@@ -1,75 +1,225 @@
...
+- for IAR tools, email to <a href="mailto:CMSIS@arm.com">CMSIS@iar.com</a>.
 
-For example, Arm uses the pack information to generate a list of available software packs. The following content will be processed in order to generate
-the web site
-(<a href="https://developer.arm.com/embedded/cmsis/cmsis-packs" target="_blank">https://developer.arm.com/embedded/cmsis/cmsis-packs</a>):
+Arm uses the pack information to generate a list of available software packs. The following content will be processed in
+order to generate the web site (<a href="https://www.keil.arm.com/packs/" target="_blank">https://www.keil.arm.com/packs/</a>):
 - Version
-- Release Notes
+- Release notes
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:169]: `<description>`  and `overview md file`.
Should we mention here that we need to register an email that the importer will send import issues to?

File: doxygen/src/pack_publish.txt [hunk 10]
```diff
@@ -89,48 +239,116 @@ element in the PDSC file. To ensure that the pack is found by a \ref cp_KeilComP
...
-\section cp_PackRenaming Rename a pack
+\subsection cp_PackRenaming Renaming a pack
 
 Sometimes, it is necessary to rename a pack (because the pack vendor and/or its name have changed for example). To ensure
 that the pack is found by a \ref cp_KeilComPack "Pack Index Service" and the tools, follow this procedure:
-- Create:
-  - a new PDSC file with the updated \elem{name} and/or \elem{vendor}, for example NewVendor.NewName.pdsc. The
+- Create a new:
+  - PDSC file with the updated \elem{name} and/or \elem{vendor}, for example NewVendor.NewName.pdsc. The
     \elem{version} should start at 1.0.0.
```
Comment (jkrech) [inline doxygen/src/pack_publish.txt:248]: Do we mention somewhere that only release quality pack versions must be published via this index?

File: doxygen/pack.dxy [hunk 11]
```diff
@@ -38,7 +38,7 @@ PROJECT_NAME           = "Open-CMSIS-Pack"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = "Version 1.7.37"
+PROJECT_NUMBER         = "Version 1.7.38"
```
Comment (jkrech) [inline doxygen/pack.dxy:41]: Do we need to  update the schema version to match the project number? Or should we rather leave the project number untouched?

**Review by jkrech — CHANGES_REQUESTED** [review]
Please see comments and questions.

File: doxygen/src/pdsc_format.txt
Comment (KeilChris) [reply doxygen/src/pdsc_format.txt:1]: The reference went to the software pack tutorial. Was removed, so I removed the references.

File: doxygen/images/pack_trinity.png
Comment (KeilChris) [reply doxygen/images/pack_trinity.png:1]: Added text to make it clearer. Projects and ref apps can reside outside a pack, so they're not core components of a pack.

File: doxygen/src/pdsc_format.txt (hunk 1 above)
Comment (KeilChris) [reply doxygen/src/pdsc_format.txt]: Corrected.

File: doxygen/src/pack_publish.txt (hunk 3 above)
Comment (KeilChris) [reply doxygen/src/pack_publish.txt]: Corrected.

File: doxygen/src/pack_publish.txt (hunk 7 above)
Comment (KeilChris) [reply doxygen/src/pack_publish.txt]: Done

File: doxygen/src/pdsc_format.txt (hunk 2 above)
Comment (KeilChris) [reply doxygen/src/pdsc_format.txt]: Reohrased.

File: doxygen/pack.dxy (hunk 11 above)
Comment (KeilChris) [reply doxygen/pack.dxy:41]: Happy to revert.

File: doxygen/src/pack_publish.txt (hunk 10 above)
Comment (KeilChris) [reply doxygen/src/pack_publish.txt:249]: Added in note on line 121.

**Review by jkrech — APPROVED** [review]
LGTM

**Pull request merged**